activerecord/lib/active_record/signed_id.rb

```diff
@@ -60,8 +60,10 @@ def find_signed(signed_id, purpose: nil)
       #   User.first.destroy
       #   User.find_signed! signed_id # => ActiveRecord::RecordNotFound
       def find_signed!(signed_id, purpose: nil)
+        raise UnknownPrimaryKey.new(self) if primary_key.nil?
+
         if id = signed_id_verifier.verify(signed_id, purpose: combine_signed_id_purposes(purpose))
-          find(id)
+          find_by! primary_key => id
```

Comment:
Originally `find` supports custom primary key.

https://github.com/rails/rails/blob/4a1b57ce70bfbbb73663b4ba68db38cbf9b6a9d4/activerecord/lib/active_record/relation/finder_methods.rb#L427-L428

https://github.com/rails/rails/blob/4a1b57ce70bfbbb73663b4ba68db38cbf9b6a9d4/activerecord/lib/active_record/relation/finder_methods.rb#L449-L458

Does `find` not work as expected?

Reply:
Sorry @kamipo my bad, I was around this code and the difference in the methods implementation caught my attention. Now that you point me to the ActiveRecord's `find` code I think this PR can be closed. Thanks